app/controllers/projects/new.js

```diff
@@ -30,80 +85,38 @@ export default class NewProjectController extends Controller {
     };
   }
 
-  searchedAddressSource = null;
-
-  taxLotsLinesLayer = {
-    id: 'pluto-line',
-    type: 'line',
-    source: 'pluto',
-    minzoom: 15,
-    'source-layer': 'pluto',
-    paint: {
-      'line-width': 0.5,
-      'line-color': 'rgba(130, 130, 130, 1)',
-      'line-opacity': {
-        stops: [
-          [
-            15,
-            0,
-          ],
-          [
-            16,
-            1,
-          ],
-        ],
-      },
-    },
+  @computed('model.rezoningArea')
+  get rezoningAreaSource() {
+    const data = this.get('model.rezoningArea');
+    return {
+      type: 'geojson',
+      data,
+    };
   }
 
-  taxLotsLabelsLayer = {
-    id: 'pluto-labels',
-    type: 'symbol',
-    source: 'pluto',
-    'source-layer': 'pluto',
-    minzoom: 15,
-    layout: {
-      'text-field': '{lot}',
-      'text-font': [
-        'Open Sans Regular',
-        'Arial Unicode MS Regular',
-      ],
-      'text-size': 11,
-    },
-    paint: {
-      'text-opacity': {
-        stops: [
-          [
-            16.5,
-            0,
-          ],
-          [
-            17.5,
-            1,
-          ],
-        ],
-      },
-      'icon-color': 'rgba(193, 193, 193, 1)',
-      'text-color': 'rgba(154, 154, 154, 1)',
-      'text-halo-color': 'rgba(152, 152, 152, 0)',
-    },
+  @computed('selectedLots.features.[]')
+  get selectedLotsSource() {
+    const selectedLots = this.get('selectedLots');
+    return {
+      type: 'geojson',
+      data: selectedLots,
+    };
   }
 
-  sources = normalizeCartoVectors([{
-    id: 'pluto',
-    type: 'cartovector',
-    minzoom: 10,
-    'source-layers': [
-      {
-        id: 'pluto',
-        sql: 'SELECT the_geom_webmercator, landuse, numfloors FROM mappluto_v1711',
-      },
-      {
-        id: 'block-centroids',
-        sql: 'SELECT the_geom_webmercator, block FROM mappluto_block_centroids',
-      },
-    ],
-  }])
+  @action toggleGeometryEditing(type) {
+    this.set('projectGeometryMode', type);
+
+    const projectGeometryMode = this.get('projectGeometryMode');
+
+    const map = this.get('mapInstance');
+    if (projectGeometryMode) {
+      map.addControl(draw, 'top-right');
+      draw.changeMode('draw_polygon');
+    } else {
+      draw.trash();
+      map.removeControl(draw);
+    }
+  }
```

Comment:
Same as comments above, it'd be good to try to encapsulate these things inside a special geometry editing component